tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -1002,6 +1003,145 @@ TYPED_TEST(
 	for (index_t i = 0; i < len; ++i)
 		EXPECT_NEAR(c[i] * b[i], a[i], get_epsilon<TypeParam>());
 }
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGMatrix_elementwise_division)
+{
+	const auto m = 2;
+	SGMatrix<TypeParam> A(m, m);
+	SGMatrix<TypeParam> B(m, m);
+
+	std::iota(A.matrix, A.matrix + m*2, 0);
+	std::iota(B.matrix, B.matrix + m*2, 1);
+
+	auto result = element_div(A, B);
+
+	for (const auto& [result, el1, el2]: zip_iterator(result ,B , A))
+    		EXPECT_NEAR(el2 / el1, result, get_epsilon<TypeParam>());
+
+	SGMatrix<TypeParam> C(m+1, m);
+	SGMatrix<TypeParam> D(m, m+1);
+	SGMatrix<TypeParam> E(m+1, m+1);
+
+	EXPECT_THROW(element_div(C, A), ShogunException);
+	EXPECT_THROW(element_div(D, A), ShogunException);
+	EXPECT_THROW(element_div(E, A), ShogunException);
+
+	EXPECT_THROW(element_div(A, C), ShogunException);
+	EXPECT_THROW(element_div(A, D), ShogunException);
+	EXPECT_THROW(element_div(A, E), ShogunException);
+}
+
+TYPED_TEST(
+    LinalgBackendEigenAllTypesTest, SGMatrix_elementwise_division_in_place)
+{
+	const auto m = 2;
+	SGMatrix<TypeParam> A(m, m);
+	SGMatrix<TypeParam> B(m, m);
+	SGMatrix<TypeParam> result(m, m);
+
+	std::iota(A.matrix, A.matrix + m*2, 0);
+	std::iota(B.matrix, B.matrix + m*2, 1);
```

Comment:
why `m*2`? surely `m*m`.. Also, you should get the matrix length with `SGMatrix::size()`

Reply:
I guess you put `m*2` because `m*2==m*m` in this case... But this is can lead to bugs in refactors, and is also unclear